packages/syft/src/syft/telemetry.py

```diff
@@ -15,10 +14,10 @@ def str_to_bool(bool_str: Optional[str]) -> bool:
 TRACE_MODE = str_to_bool(os.environ.get("TRACE", "False"))
 
 
-def setup_tracer() -> Any:
+def setup_tracer():
     if not TRACE_MODE:
 
-        def noop(func: Any) -> Any:
+        def noop(func):
```

Comment:
Any reason to remove typing ?

Reply:
This typing is not accurate. The correct one would be

```py
T = TypeVar("T")

def noop(func: T) -> T:
    ...
```

If you don't type it as `Any`, type inference would figure out that `noop` returns the same type as `func`. Earlier, because `serializable` and `instrument` decorators as `Any` were given `Any` as the return type, anything with these decorator would lose its type and become `Any`. This is the reason why we didn't have type or code completion for `BaseStash`, `PartitionKey`, `QueryKey`, ... Either use type parameters to type decorators correctly, or just don't give them types and let type inference figures it out.

Reply:
Removing `Any` here and `setup_tracer` already makes things work so I just left it at that. Type inference for decorators is not implemented perfectly either so I was just going with the simplest solution that works here.

Reply:
Explicit typing re-added in #7483.